src/TorchSharp/Dataset.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Collections.Generic;
+
+namespace TorchSharp
+{
+    public static partial class torch
+    {
+        public static partial class utils
+        {
+            public static partial class data
+            {
+                public abstract class Dataset : IDisposable
+                {
+                    public virtual void Dispose()
+                    {
+                    }
+
+                    public abstract int Count { get; }
```

Comment:
Since data sets can be enormous, Count and GetTensor (and anyone calling them) should probably use 'long' instead of 'int'. It's not often that you need more than 32-bit int for indices, but I think this is one of those rare situations.

The same goes for all other places in the code where the integer is an index into the data set.